bookkeeper-server/src/main/java/org/apache/bookkeeper/conf/ServerConfiguration.java

```diff
@@ -2023,7 +2027,49 @@ public ServerConfiguration setNettyMaxFrameSizeBytes(int maxSize) {
     }
 
     /**
-     * Gets the minimum safe Usable size to be available in index directory for Bookie to create Index File while replaying 
+     * Sets that whether the http server can start along with auto-recovery service
+     *
+     * @param enabled
+     *            - true if need to start http server with auto-recovery
+     * @return ServerConfiguration
+     */
+    public ServerConfiguration setHttpServerEnabled(boolean enabled) {
+        setProperty(HTTP_SERVER_ENABLED, enabled);
+        return this;
+    }
+
+    /**
+     * Get whether the http server start with auto-recovery service or not
+     *
+     * @return true - if http server should start with auto-recovery service
+     */
+    public boolean isHttpServerEnabled() {
+        return getBoolean(HTTP_SERVER_ENABLED, false);
+    }
+
+    /**
+     * Set Http server port listening on
+     *
+     * @param port
+     *          Port to listen on
+     * @return server configuration
+     */
+    public ServerConfiguration setHttpServerPort(int port) {
```

Comment:
@sijie  @yzang I apologize I was not clear in my comment.
I think that there are two opposite ways:
1) the implementation handles all of the parameters
so Vertx, TwitterServer impl can have their own "port" parameter, there will be no 'standard' http port configuration parameter

2) we define that there are the common 3 parameters for every httpserver implementation:
- port (defaults to xxxx)
- binding address (defaults to 0.0.0.0)
- secure: true|false (default to false, maybe not implemented by vertx/twitter server)

maybe we can add 'secure' in a future time, but if we make configurable the port, let's add at least the 'binding address'.

if we make address/port/secure configurable with 'standard' attributes it will be easy to recompose the URLs of the API just by reading the configuration file